core/dbt/config/project.py

```diff
@@ -568,11 +570,6 @@ def from_project_root(
         ) = package_and_project_data_from_root(project_root)
         selectors_dict = selector_data_from_root(project_root)
 
-        if "flags" in project_dict:
```

Comment:
This feels ominous. Was there a specific reason we didn't want to pull `flags` through here? Was it just that we didn't want flags to appear both here and as `ProjectFlags`? If so, do we need to consider where `ProjectFlags` is used?

Reply:
By adding the flags in this way, we are not considering CLI arguments nor environment variables. This is not the same as the global flags that get used elsewhere. I want to be explicit here because this may cause confusion. It's also worth considering that we do not want to support setting behavior flags dynamically (e.g. via CLI or environment variables).

That begs the questions:

1. Should this actually be the global flags, and not just the project flags?
2. Should this be called something else?
3. Should we have both the global flags and project flags (could be confusing)?

Reply:
It looks like this happened because they were always separate and just got renamed here: https://github.com/dbt-labs/dbt-core/pull/9289.

Reply:
@gshank Do you know the history here?

Reply:
@mikealfare Not sure, but should we consider calling this collection "behavior flags"? Would it be possible to type it more strongly than Dict[str, Any]?

Reply:
As discussed in Slack, this does refer to all flags, so the name and type make some sense.